app/src/main/java/com/github/libretube/ui/dialogs/SubmitSegmentDialog.kt

```diff
@@ -0,0 +1,86 @@
+package com.github.libretube.ui.dialogs
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import android.util.Log
+import android.widget.ArrayAdapter
+import androidx.fragment.app.DialogFragment
+import androidx.lifecycle.lifecycleScope
+import com.github.libretube.BuildConfig
+import com.github.libretube.R
+import com.github.libretube.api.RetrofitInstance
+import com.github.libretube.databinding.DialogSubmitSegmentBinding
+import com.github.libretube.extensions.TAG
+import com.github.libretube.extensions.toastFromMainDispatcher
+import com.github.libretube.helpers.PreferenceHelper
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import java.lang.Exception
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class SubmitSegmentDialog(
+    private val videoId: String,
+    private val currentPosition: Long,
+    private val duration: Long?
+) : DialogFragment() {
+    private lateinit var binding: DialogSubmitSegmentBinding
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        binding = DialogSubmitSegmentBinding.inflate(layoutInflater)
+
+        binding.startTime.setText((currentPosition.toFloat() / 1000).toString())
+
+        val categoryNames = resources.getStringArray(R.array.sponsorBlockSegmentNames)
+        binding.segmentCategory.adapter = ArrayAdapter(
+            requireContext(),
+            R.layout.dropdown_item,
+            categoryNames
+        )
+
+        return MaterialAlertDialogBuilder(requireContext())
+            .setTitle(getString(R.string.sb_create_segment))
+            .setView(binding.root)
+            .setPositiveButton(R.string.okay, null)
+            .setNegativeButton(R.string.cancel, null)
+            .show()
+            .apply {
+                getButton(DialogInterface.BUTTON_POSITIVE).setOnClickListener {
+                    requireDialog().hide()
+
+                    lifecycleScope.launch {
+                        submitSegment()
+                        dismiss()
+                    }
+                }
+            }
```

Comment:
```suggestion
            .setPositiveButton(R.string.okay) { _, _ ->
                requireDialog().hide()

                lifecycleScope.launch {
                    submitSegment()
                    dismiss()
                }
            }
            .setNegativeButton(R.string.cancel, null)
            .show()
```

Reply:
This doesn't work, because we get a `CoroutineCancelationException` then, as the life cycle ends automatically after a short time when clicking the dialog button. We already do the same work around as here at other places, e.g. in https://github.com/libre-tube/LibreTube/blob/b1cccdd935c3de3ad3136b9e30b809f82b22363f/app/src/main/java/com/github/libretube/ui/dialogs/RenamePlaylistDialog.kt#L40.